applications/app/templates/serviceWorker.scala.js

```diff
@@ -161,16 +162,13 @@ self.addEventListener('push', function(event) {
             .then(function (json) {
 
                if(json.status === "ok")
-                    var messages = json.messages;
-
-                    messages.forEach( function(message) {
-                        var data = {topic: message.topic};
-                        self.registration.showNotification(message.title, {
-                            body: message.body,
-                            icon: '@{JavaScript(Static("images/favicons/114x114.png").path)}',
-                            tag: message.title,
-                            data: data
-                        });
+                    var message = json.messages.slice(-1)[0];
```

Comment:
Can you guarantee json will always contain messages, will always contain the number etc or is it worth putting some safeguards around this? Also a comment for why it's needed would be useful :)